File: backend/routers/Post.router.go
package routers

import (
	"github.com/gin-gonic/gin"
	"github.com/go-redis/redis/v8"
	"github.com/jmoiron/sqlx"
	"github.com/lxndrrud/blog-project/controllers"
	"github.com/lxndrrud/blog-project/infrastructure"
	"github.com/lxndrrud/blog-project/repositories"
	"github.com/lxndrrud/blog-project/services"
	"github.com/lxndrrud/blog-project/utils"
)

func setupPostRouter(router *gin.RouterGroup, db *sqlx.DB, redisConn *redis.Client) {
	postRepo := repositories.NewPostsRepo(db)
	userRepo := repositories.NewUserRepo(db)
	permissionInfra := infrastructure.NewPermissionInfra(db, redisConn)
	fileProcessor := utils.NewFileProcessor()
	permissionChecker := utils.NewPermissionChecker()

	postService := services.NewPostService(
		postRepo,
		userRepo,
		permissionInfra,
		fileProcessor,
		permissionChecker,
	)

	controller := controllers.NewPostController(postService)

	router.GET("/userPosts", controller.GetUserPosts)
	router.GET("/actual", controller.GetApprovedPosts)
	router.GET("/needToApprove", controller.GetPostsNeedToApprove)
	router.GET("/get", controller.GetApprovedPost)
	router.GET("/needToApprove/get", controller.GetPostNeedToApprove)
	router.POST("/new", controller.CreatePost)
	router.POST("/approve", controller.ApprovePost)
	router.POST("/reject", controller.RejectPost)
	router.DELETE("/delete", controller.DeletePost)
}

File: backend/src/routes/Post.router.ts
import { Router } from "express";
import { postController } from "../container";

export const PostRouter = Router()

PostRouter.get('/userPosts', postController.getUserPosts.bind(postController))
PostRouter.get('/actual', postController.getApprovedPosts.bind(postController))
PostRouter.get('/needToApprove', postController.getPostsNeedToApprove.bind(postController))
PostRouter.get('/get', postController.getApprovedPost.bind(postController))
PostRouter.get('/needToApprove/get', postController.getApprovedPost.bind(postController))
PostRouter.post('/new', postController.createPost.bind(postController))
PostRouter.post('/approve', postController.approvePost.bind(postController))
PostRouter.post('/reject', postController.rejectPost.bind(postController))
PostRouter.delete('/delete', postController.deletePost.bind(postController))
